llvm/test/Transforms/CodeGenPrepare/X86/split-dbg.ll

```diff
@@ -0,0 +1,49 @@
+; RUN: opt -passes='require<profile-summary>,function(codegenprepare)' -S -mtriple=x86_64 < %s \
+; RUN: | FileCheck %s
+
+;; Check debug locations are propagated onto new PHIs.
+
+; CHECK: .split:
+; CHECK-NEXT: %merge = phi i32 [ poison, %while.body ], [ %dest.sroa.clone, %while.body.clone ], !dbg [[DBG:!.*]]
```

Comment:
Worthwhile to check that it's the correct DILocation too?

Reply:
I didn't think it was needed because there's only one !dbg on the input IR - but it couldn't hurt, done.